.github/workflows/build.yaml

```diff
@@ -70,3 +77,17 @@ jobs:
           platforms: linux/amd64,linux/arm64,linux/arm64/v8
           push: true
           tags: quay.io/abhishekvrshny/propeller:latest
+      - name: Docker Image Metadata
```

Comment:
should this be done before pushing the image?

Reply:
no i dont think so , the docker/metadata-action is typically used to annotate or add metadata to the Docker image that has already been built and pushed. If you run this step before the image is pushed to the registry (in this case, quay.io), the metadata won't actually be associated with the pushed image.

Reply:
i think it is correctly ordered but if wish to discuss further i can provide more elaborate explanation as well

Reply:
The official documentation of `metadata-action` suggests otherwise. The `build-push-action` step comes later and uses the tags and labels from the `metadata-action` step. Am I missing something here? Ref: https://github.com/docker/metadata-action?tab=readme-ov-file#basic

Reply:
I looked upon the docs, the difference lies in approach both will work but my current implementation is rather future proof as it avoids overhead in metadata calculation , avoids potential redundancy of tags and labels and decouples build and metadata layer for clearer implementation. If still you would suggest, I can revert to official implementation but I think this one works just fine

Reply:
In the current approach, from what I understand, the labels and tags would be applied to the image but locally. Does this action update the already uploaded image to quay.io with the newly added tags and labels again? For this, it would need to push the image again to quay.io, but I couldn't find any such reference in the docs or codebase of this action here https://github.com/docker/metadata-action/tree/master/src

Reply:
The `images: quay.io/abhishekvrshny/propeller` input is just used to define the base name for the tags. It is not used to push the updated tags and labels. https://github.com/docker/metadata-action?tab=readme-ov-file#images-input

Reply:
That is correct, the official implementation would update the tags and labels with the new added tags and labels. Current implementation I doubt would

> In the current approach, from what I understand, the labels and tags would be applied to the image but locally. Does this action update the already uploaded image to quay.io with the newly added tags and labels again? For this, it would need to push the image again to quay.io, but I couldn't find any such reference in the docs or codebase of this action here https://github.com/docker/metadata-action/tree/master/src

Reply:
#### in build.yaml:
    
      - name: Docker Image Metadata
        id: meta
        uses: docker/metadata-action@v4
        with:
          images: quay.io/abhishekvrshny/propeller
          tags: |
            latest
            ${{ github.sha }}
          labels: |
            org.opencontainers.image.source=${{ github.repository }}
            org.opencontainers.image.revision=${{ github.sha }}
            org.opencontainers.image.created=${{ steps.date.outputs.date }}

      - name: Build and push Docker image
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile
          platforms: linux/amd64,linux/arm64,linux/arm64/v8
          push: true
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
          
 The above refractoring would solve the issue in the current implementation, but I think the decoupling of labels of tags for a moment seems like a cleaner approach

Reply:
let me know if I should proceed with this or keep it as it is for now

Reply:
Having metadata action post push action doesn't solve anything. We better don't have this step in that case. I am unsure about what you mean by decoupling. Building an image, associating tags to it, and pushing it tothe  image registry is 1 process and has multiple steps which are sequential and the output of one is used as the input of another. I don't see any challenge with this. If we have to keep the metadata step then it only helps if done before push step.

Reply:
got it, i seemed to have kept that for better logging of information, would remove it now

Reply:
done with the changes, i think this lgtm